airship-adapter/src/main/java/com/gimbal/airship/AirshipAdapter.java

```diff
@@ -246,6 +220,99 @@ public synchronized static AirshipAdapter shared(@NonNull Context context) {
         return instance;
     }
 
+    public void unloadQueuedEvents() {
+        for (CachedVisit cachedVisit: cachedVisits) {
+            createAirshipEvent(cachedVisit.visit, cachedVisit.regionEvent);
+        }
+        cachedVisits = new LinkedList<>();
+    }
+
+    private void createAirshipEvent(Visit visit, int regionEvent) {
```

Comment:
Pretty sure you can refactor this so that code isn't duplicated with the `PlaceEventListener` code.